### PR TITLE
initial commit of adding psycopg2

### PR DESCRIPTION
**Pull request by msarahan**
This is one that people have complained about OpenSSL getting out of date, requiring us to rebuild it.  I think it was just a really old version of OpenSSL, before the switch to 1.0.2.  These recipes should be pretty low maintenance.

Note that Postgresql here has Windows - this is new.  I added it for this recipe.  We don't have this in our repo.  I haven't tested it much.  I welcome feedback.

### CODE REVIEW

**Comment by ocefpaf** [comment]
Awesome! That one was in my list :wink:

**Comment by msarahan** [comment]
This recipe currently omits OpenSSL from the mac builds.  I tried to point it at the right libraries, but it just will not take them, and the error messages are inscrutable.  If anyone knows Mac and wants to have a look, please have at it.

File: recipes/perl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package:
+  name: perl
+  version: 5.20.3.1
+
+source:
+  fn:  strawberry-perl-5.20.3.1-32bit-portable.zip [win]
```
Comment (pelson) [inline recipes/perl/meta.yaml]: What is the expected behaviour when on osx and linux? Use the system perl?

**Comment by pelson** [comment]
```
%CMD_IN_ENV% conda-build-all recipes --matrix-conditions "numpy >=1.8" "%PY_CONDITION%"
Using default MSVC build environment for 32 bit architecture
Executing: conda-build-all recipes --matrix-conditions "numpy >=1.8" "python >=2.7,<3"
Using Anaconda Cloud api site https://api.anaconda.org 
Fetching package metadata: ...... 
Resolving distributions from 4 recipes...  
Traceback (most recent call last): 
  File "C:\Miniconda\Scripts\conda-build-all-script.py", line 9, in <module> 
    load_entry_point('conda-build-all==0.9.0', 'console_scripts', 'conda-build-all')() 
  File "C:\Miniconda\lib\site-packages\conda_build_all\cli.py", line 85, in main 
    b.main() 
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 195, in main
    all_distros = self.compute_build_distros(index, recipe_metas)
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 176, in compute_build_distros 
    self.matrix_conditions)
  File "C:\Miniconda\lib\site-packages\conda_build_all\resolved_distribution.py", line 97, in resolve_all
    cases = sorted(vn_matrix.special_case_version_matrix(meta, index)) 
  File "C:\Miniconda\lib\site-packages\conda_build_all\version_matrix.py", line 224, in special_case_version_matrix
    raise NotImplementedError('PERL version matrix not yet implemented.')
NotImplementedError: PERL version matrix not yet implemented. 
```

Comes from https://github.com/SciTools/conda-build-all/blob/master/conda_build_all/version_matrix.py#L223.

The only reason that wasn't implemented was through lack of an understanding of what was needed. @msarahan - I'm guessing your perl foo is >> than mine, so if you wanted to take a stab at implementing it in conda-build-all, I'm :+1: with that. Otherwise, looks like it might be simply pinning the perl version to the full (major, minor, patch) version as per https://github.com/conda/conda-build/blob/a70f02d5a7d58fa27e03596e1bf16fb27dbc7254/conda_build/metadata.py#L317.

File: recipes/perl/meta.yaml (hunk 1 above)
Comment (msarahan) [inline recipes/perl/meta.yaml]: yes.  That's been a generally safe assumption for me, but it is an assumption that may not be true.  It is true on OS X, and on our docker build image for linux.

**Comment by msarahan** [comment]
My perl is actually not great.  I can write regexes, but I don't really use it in anger.

I have pinned the version for now.

File: recipes/perl/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/perl/meta.yaml]: I was proposing to add a Perl recipe for Mac and Linux anyways (before I knew about this). Proposed changes to this PR can be seen here ( https://github.com/msarahan/staged-recipes/pull/1 ).

**Comment by jakirkham** [comment]
So, I think we can drop the Perl from here as it is being released as a feedstock ATM. Though that will take a bit as the CIs are backed up right now.

**Comment by msarahan** [comment]
Yeah, good call.  I'll get rid of it.  This is still not working because of the conda-build-all stuff for Perl.  The hard-coding didn't help.  @pelson any ideas?  I'd like to fix conda-build-all, but I can't justify the time right now.

**Comment by pelson** [comment]
> @pelson any ideas? I'd like to fix conda-build-all, but I can't justify the time right now.

I'll try to take a look tomorrow. Shouldn't take too long.

**Comment by jakirkham** [comment]
BTW Perl is now available in conda-forge for all platforms. So, it is safe to remove from your PR.

**Comment by pelson** [comment]
I've just pulled together the necessary changes for a perl matrix to be supported (and also R matrix). https://github.com/SciTools/conda-build-all/pull/27

I'll get on and merge + release the changes asap.

**Comment by jakirkham** [comment]
How will this translate into the build matrix? Do you have some logic for that somewhere or is it still WIP?

**Comment by pelson** [comment]
> How will this translate into the build matrix? Do you have some logic for that somewhere or is it still WIP?

Good question. I know I will need to update https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/templates/appveyor.yml.tmpl#L1, but suspect there will be a couple of bitty changes needed in conda-smithy.

**Comment by ocefpaf** [comment]
I am clueless on Windows, but I saw some updates here that might help with those failures, no?

**Comment by pelson** [comment]
I think this PR could probably do with being rebased - a lot has changed in 25 days 😄

**Comment by ocefpaf** [comment]
@msarahan I know you are busy, so do you mind if I give this PR a try?

**Comment by msarahan** [comment]
Sure, knock yourself out!  Let me know if I can help.

**Comment by ocefpaf** [comment]
Thanks. See #291

**Pull request closed without merging**